### DISCUSSION TITLE
Generated hooks from OPENAPI schema

### DISCUSSION THREAD

**Discussion by Patriksafar** [TanStack/query#5071] [Q&A] [answered]
Hi,

is there any tool I can use or example how to use OpenAPI schema to generate hooks for `@tanstack/react-query`?

**Comment by louis-young** [comment, marked as answer]
Hi 👋 

Yeah, there are various tools available. Personally, I like Orval 🙂 

https://orval.dev/guides/react-query

**Reply by Patriksafar** [reply]
Thank you for suggestion @louis-young ! 🙂 I will have a look on that.
